init.lua

```diff
@@ -82,17 +87,18 @@ local function calculateVisibilityPolygon(originX, originY, radius, polygons)
       local x = polygon[i]
       local y = polygon[i+1]
       local a1, a2 = vector.sub(x, y, originX, originY)
+
       local angleA = math.atan2(a2, a1)
-      local angleB = angleA + 0.0001
-      local angleC = angleA - 0.0001
+
+      local angles = {
+      	angleA,
+      	angleA + 0.0001,    --angleB
+      	angleA - 0.0001,    --angleC
+      }
 
       -- Go through all 3 angles as rays cast from originX, originY
       for j=1,3 do
-        local angle
-        if j == 1 then angle = angleA end
-        if j == 2 then angle = angleB end
-        if j == 3 then angle = angleC end
-
+        local angle = angles[j]
```

Comment:
Hmm, the original syntax was there to avoid creating a temp table, as this is in a loop and performed potentially many times per frame, especially if there are multiple moving lights or lots of updates to the geometry. (as this is a library, in my own game code I wouldn't worry about the extra table)

Reply:
I think that the temp table for only 3 values is better than 3 if-statements. Iirc it is constant time to access a value in an array using number index as opposed to key index.

Reply:
Also, we are using an array to hold 3 values of type double/float, isnt that the same size as with creating 3 variables? after all, the size of the array and the 3 variables are the same.

Reply:
Ah right, it's not about access-time or memory, it's about GC being invoked. (for which temporary tables are the worst) This is why libraries like vector-light explicitly avoid temp tables. In theory all the operations are the same, but creating temporary tables is a GC killer. Your thoughts on that?

Reply:
Hmm that i do not know for sure 🤔 but i think the gc wouldnt spike and be a problem since the memory footprint of a table with  a constant number of 3 values is very low. Maybe luajit will do some magic with it?

Perhaps better is to do:

```lua
--file scope
local _angles = {}

--inside the function
angles[1] = --value1
angles[2] = --value2
angles[3] = --value3
```

This way there is no temp table.

Reply:
Hmm, I like this idea! :+1: Would you like to edit your PR to this? (just add a small comment to indicate something like "This table is defined once outside of the function scope to avoid creating a new table on every call")

Reply:
Alright! Will do first thing in the morning. Thanks

Reply:
Nice!